src/apps/api/tests/test_phases.py

```diff
@@ -46,3 +49,43 @@ def test_normal_user_cannot_re_run_whole_phase(self):
         resp = self.client.get(reverse('phases-rerun_submissions', kwargs={'pk': self.phase.id}))
         assert resp.status_code == 403, 'Did not raise permission denied and should have'
         assert self.phase.submissions.count() == 4
+
+
+class CompetitionPhaseMigrationValidation(TestCase):
+    def setUp(self):
+        self.user = UserFactory(username='test')
+        self.competition = CompetitionFactory(created_by=self.user, id=1)
+        self.phase = PhaseFactory(competition=self.competition)
+        self.leaderboard = LeaderboardFactory(competition=self.competition)
+
+    def serialize_and_validate_data(self):
+        serializer = CompetitionSerializer(self.competition)
+        data = serializer.data
+
+        with pytest.raises(ValidationError) as exception:
+            serializer = CompetitionSerializer(data=data)
+            serializer.is_valid(raise_exception=True)
+
+        return exception
+
+    def test_phase_is_valid(self):
+        self.phase.auto_migrate_to_this_phase = False
+        self.phase.save()
+        exception = self.serialize_and_validate_data()
+
+        assert ("'phase:'" not in str(exception.value))
+
+    def test_phase_serializer_auto_migrate_on_first_phase(self):
+        self.phase.auto_migrate_to_this_phase = True
+        self.phase.save()
+        exception = self.serialize_and_validate_data()
+
+        assert ("You cannot auto migrate in a competition with one phase" or
+                "You cannot auto migrate to the first phase of a competition" in str(exception.value))
```

Comment:
This is the equivalent of
```python
assert "You cannot auto migrate in a competition with one phase" in str(exception.value))
```
for the logic you want, either:
```python
assert "You cannot auto migrate in a competition with one phase" in str(exception.value) \
       or "You cannot auto migrate to the first phase of a competition" in str(exception.value)
```
or
```python
errors = ["You cannot auto migrate in a competition with one phase", "You cannot auto migrate to the first phase of a competition"]
assert any(error in str(exception.value) for error in errors)

```

Reply:
Ah right! I make this mistake too often, thanks Jimmy!